lib/src/objdetect/objdetect_async.dart

```diff
@@ -0,0 +1,917 @@
+              ref,
+              img.ref,
+              hitThreshold,
+              winStride.cvd.ref,
+              padding.cvd.ref,
+              callback,
+            ), (c, foundLocations, searchLocations) {
+      return c.complete(
+        (
+          VecPoint.fromPointer(foundLocations.cast<cvg.VecPoint>()),
+          VecPoint.fromPointer(searchLocations.cast<cvg.VecPoint>())
+        ),
+      );
+    });
+    return rval;
+  }
+
+  Future<VecRect> detectMultiScaleAsync(
+    InputArray image, {
+    double hitThreshold = 0,
+    int minNeighbors = 3,
+    (int, int) winStride = (0, 0),
+    (int, int) padding = (0, 0),
+    double scale = 1.05,
+    double groupThreshold = 2.0,
+    bool useMeanshiftGrouping = false,
+  }) async {
+    final rval = cvRunAsync<VecRect>(
+        (callback) => CFFI.HOGDescriptor_DetectMultiScaleWithParams_Async(
+              ref,
+              image.ref,
+              hitThreshold,
+              winStride.cvd.ref,
+              padding.cvd.ref,
+              scale,
+              groupThreshold,
+              useMeanshiftGrouping,
+              callback,
+            ), (c, rects) {
+      return c.complete(VecRect.fromPointer(rects.cast<cvg.VecRect>()));
+    });
+    return rval;
+  }
+
+  static Future<VecFloat> getDefaultPeopleDetectorAsync() async {
+    final rval =
+        cvRunAsync<VecFloat>(CFFI.HOG_GetDefaultPeopleDetector_Async, (c, v) {
+      return c.complete(VecFloat.fromPointer(v.cast<cvg.VecFloat>()));
+    });
+    return rval;
+  }
+
+  static Future<VecFloat> getDaimlerPeopleDetectorAsync() async {
+    final rval = cvRunAsync<VecFloat>(
+        CFFI.HOGDescriptor_getDaimlerPeopleDetector_Async, (c, v) {
+      return c.complete(VecFloat.fromPointer(v.cast<cvg.VecFloat>()));
+    });
+    return rval;
+  }
+
+  Future<int> getDescriptorSizeAsync() async {
+    final rval = cvRunAsync<int>(
+        (callback) => CFFI.HOGDescriptor_getDescriptorSize_Async(ref, callback),
+        (c, p) {
+      final rval = p.cast<ffi.Size>().value;
+      calloc.free(p);
+      return c.complete(rval);
+    });
+    return rval;
+  }
+
+  Future<double> getWinSigmaAsync() async {
+    final rval = cvRunAsync<double>(
+        (callback) => CFFI.HOGDescriptor_getWinSigma_Async(ref, callback),
+        (c, p) {
+      final rval = p.cast<ffi.Double>().value;
+      calloc.free(p);
+      return c.complete(rval);
+    });
+    return rval;
+  }
+
+  Future<(VecRect rectList, VecDouble weights)> groupRectanglesAsync(
+    VecRect rectList,
+    VecDouble weights,
+    int groupThreshold,
+    double eps,
+  ) async {
+    final rval = cvRunAsync<(VecRect, VecDouble)>(
+        (callback) => CFFI.HOGDescriptor_groupRectangles_Async(
+              ref,
+              rectList.ref,
+              weights.ref,
+              groupThreshold,
+              eps,
+              callback,
+            ), (c, _) {
+      return c.complete((rectList, weights));
+    });
+    return rval;
+  }
```

Comment:
Here, rectList and weights are input/output, so they will be modified internally, thus a CvCallback_0/cvRunAsync0 is needed